src/client/Microsoft.Identity.Client/PoP/PoPAuthenticationScheme.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Json.Linq;
+
+namespace Microsoft.Identity.Client.PoP
+{
+    internal class PoPAuthenticationScheme : IAuthenticationScheme
+    {
+        private static readonly DateTime s_jwtBaselineTime = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        private readonly Uri _uriToBind;
+        private readonly IPoPCryptoProvider _popCryptoProvider;
+
+        /// <summary>
+        /// Creates POP tokens, i.e. tokens that are bound to an HTTP request and are digitally signed.
+        /// </summary>
+        /// <param name="uriToBind">Uri describing the HTTP request</param>
+        /// <param name="popCryptoProvider">Crypto used to sign the POP token</param>
+        public PoPAuthenticationScheme(Uri uriToBind, IPoPCryptoProvider popCryptoProvider)
```

Comment:
No HTTP verb?

Reply:
We also considered passing the HttpRequest (which would contain both the URI and the Http Verb. Would that make sense?

Reply:
Ack. I remember that we wanted to have the verb as being optional, so this is the minimum required to get this to work.